### DISCUSSION TITLE
signIn and signOut components redirect to another button rather straight to authorization page.

### DISCUSSION THREAD

**Discussion by hanpham32** [nextauthjs/next-auth#11042] [Help] [answered]
I want the signIn to redirect straight to the Github authentication page rather to another button. Is there a way to do that?

**Comment by k-taro56** [comment, marked as answer]
Hi,

You should add `github` to the argument of the `signIn` function like `signIn("github")`.

https://authjs.dev/reference/nextjs#signin-2

```ts
import { signIn } from "../auth"
 
export default function Layout() {
 return (
  <form action={async () => {
    "use server"
    await signIn("github")
  }}>
   <button>Sign in with GitHub</button>
  </form>
)
```

Thank you.
